src/Controller/TideApiController.php

```diff
@@ -351,9 +351,8 @@ private function resolveAliasPath(Request $request, $path, $site, $cid, array &$
       // Update the response.
       $code = $event->getCode();
       $json_response = $event->getJsonResponse();
-      if ($event->isOk()) {
-        // @TODO: Entity is not always a node.
-        $url = Url::fromRoute('entity.node.canonical', ['node' => $json_response['data']['attributes']['entity_id']]);
+      if ($event->isOk() && $event_entity) {
+        $url = Url::fromRoute('entity.' . $event_entity->getEntityTypeId() . '.canonical', [$event_entity->getEntityTypeId() => $json_response["data"]["entity_id"]]);
```

Comment:
@anthony-malkoun `node` and `media` are there but be careful with the route `entity.ENTITY_TYPE.canonical` as it's not always available and Uri::fromRoute may throw an exception.

Reply:
@sonnykt Would this work to avoid an exception being thrown?
```suggestion
      $entity_type_id = $event_entity->getEntityTypeId();
      
      $safe_routes = [
        'node',
        'media'
      ];

      if (!in_array($entity_type_id, $safe_routes)) {
        if (RouteProviderInterface::getRoutesByNames(['entity.' . $entity_type_id . '.canonical'])) {
          if ($event->isOk() && $event_entity) {
            $url = Url::fromRoute('entity.' . $entity_type_id . '.canonical', [$event_entity->getEntityTypeId() => $json_response["data"]["entity_id"]]);
            // Cache the response with the same tags with the entity.
            $cache_entity = $this->apiHelper->findEntityFromUrl($url);
            $cached_route_data = [
              'json_response' => $json_response['data'],
              'uri' => $url->toUriString(),
              'id' => $entity ? $entity->uuid() : NULL,
            ];
    
            // Cache the response.
            if ($cache_entity) {
              $this->cache('data')
                ->set($cid, $cached_route_data, Cache::PERMANENT, $cache_entity->getCacheTags());
            }
          }
          // Something set the Event to failure.
          else {
            unset($json_response['data']);
          }
        }
      }
```